src/components/FilterCard.vue

```diff
@@ -83,42 +79,48 @@ export default {
       default: true,
     },
   },
-  data() {
-    return {
-      filter: {
-        value: '',
-        input: '',
-        wildcard: true,
-      },
-    }
+  setup() {
+    const { filter_input, filter_value, filter_wildcard, page_size, page_number, clearFilter } =
+      useQueryParams()
+    return { filter_input, filter_value, filter_wildcard, page_size, page_number, clearFilter }
   },
   computed: {
     isWildcardOption() {
-      return this.filterOptions.filter(({ value }) => value == this.filter.value)[0]?.wildcard
+      return this.filterOptions.filter(({ value }) => value == this.filter_value)[0]?.wildcard
+    },
+  },
+  watch: {
+    /*
+      page_number and page_size:
+      We ensure the page is paginated before fetching data.
+      This catches an edge case where the watchers are still active when navigating
+      between paginated and unpaginated routes
+    */
+    page_size() {
+      this.$route.meta?.paginated ? this.getData() : ''
+    },
+    page_number() {
+      this.$route.meta?.paginated ? this.getData() : ''
     },
   },
   methods: {
     async getFilteredData() {
-      let searchValue = this.filter.input
-      if (this.isWildcardOption && this.filter.wildcard) {
-        // If wildcard is selected, add it to the search string
-        searchValue += ',wildcard'
+      if (this.page_number == 1) {
+        await this.getData()
```

Comment:
Yes its a good point. Couple things to bear in mind as to why I have done it this way:

- We don't want to refetch every time a character is typed into the filter as that will cause a lot of unnecessary requests.
- The filter needs to be reset to page: 1 when its searched as the new results may have less pages that the existing results which would break the page until reset.

That said I will take another look and see what I can do as I think you are right that there is probably a nicer way to encapsulate all the data requests. Open to suggestions.

Reply:
Yep, I agree, those points are very valid. 
I don't whether it make sense, but to have a  state inside composable that depends on query params , which watches over values to do fetching.